pool/web/handler/contact_signup.ml

```diff
@@ -67,14 +49,43 @@ let sign_up_create req =
        let* { Pool_context.Tenant.tenant; _ } =
          Pool_context.Tenant.find req |> Lwt_result.lift
        in
-       let* events =
+       let* create_contact_events =
          let open CCResult.Infix in
          Command.SignUp.(
            decode urlencoded
            >>= handle ~tags ?allowed_email_suffixes tenant preferred_language)
-         >>= (fun e -> Ok (remove_contact_event @ e))
          |> Lwt_result.lift
        in
+       let* email_address =
+         Sihl.Web.Request.urlencoded Field.(Email |> show) req
+         ||> CCOption.to_result ContactSignupInvalidEmail
+         >== Pool_user.EmailAddress.create
+       in
+       let%lwt existing_user =
+         Service.User.find_by_email_opt
+           ~ctx:(Pool_tenant.to_ctx database_label)
+           (Pool_user.EmailAddress.value email_address)
+       in
+       let* events =
+         match existing_user with
+         | None -> Lwt_result.return create_contact_events
+         | Some user ->
+           (match Service.User.is_admin user with
+            | true -> Lwt_result.return []
+            | false ->
+              email_address
+              |> Contact.find_by_email database_label
+              ||> (function
+              | Ok contact ->
```

Comment:
```suggestion
              | Ok contact when contact.Contact.user.Sihl_user.confirmed ->
```

- add `| Ok _` to the second option
- may check to add the `when` already in the most outer `match ... with` statement

Reply:
@mabiede done